extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -57,9 +95,80 @@ export class AmpSlideScroll extends BaseCarousel {
 
   /** @override */
   hasPrev() {
+    return this.slideIndex_ > 0;
   }
 
   /** @override */
   hasNext() {
+    return this.slideIndex_ < this.slides_.length - 1;
+  }
+
+  /** @override */
+  goCallback(dir, unusedAnimate) {
+    if (this.slideIndex_ != null) {
+      if ((dir == DIR_FORWARD && this.hasNext()) ||
+          (dir == DIR_REVERSE && this.hasPrev())) {
+        this.mutateElement(() => {
+          this.showSlides_(this.slideIndex_ + dir);
+        }, this.slidesContainer_);
+      }
+    }
+  }
+
+  /**
+   * Makes the slide corresponding to the given index and the slides surrounding
+   *    it available for display.
+   * @param {number} newindex Index of the slide to be displayed.
+   * @private
+   */
+  showSlides_(newindex) {
+    const noOfSlides = this.noOfSlides_;
+    if (newindex < 0 ||
+        newindex >= this.noOfSlides_ ||
+        this.slideIndex_ == newindex) {
+      return;
```

Comment:
The current protocol allows looping. So let's support or file a bug to support - but it's part of the current promised protocol.